### ISSUE TITLE
Comments between case lines trigger no-fallthrough

### ISSUE THREAD

**Issue by chris-morgan** [eslint/eslint#10181]
* **ESLint Version:** all (e.g. 4.19.1)
* **Node Version:** all
* **npm Version:** all

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration (just eslint:recommended)</summary>

```js
module.exports = {
    "extends": "eslint:recommended",
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
switch ('foo') {
	case 'foo':
	case 'bar':
	// Hmm.
	case 'baz':
}
```

```bash
eslint x.js
```

**What did you expect to happen?**

No errors: the comment is purely a remark on the next `case` and does not constitute a case body.

**What actually happened? Please include the actual, raw output from ESLint.**

```
/tmp/x.js
  5:2  error  Expected a 'break' statement before 'case'  no-fallthrough

✖ 1 problem (1 error, 0 warnings)
```

**Further remarks:**

A slightly more realistic example of this pattern:

```js
function authorOfCharacter(name) {
    switch (name) {
        // Names from http://some.example/
        case 'Julius Handford':
        case 'John Rokesmith':
        case 'John Harmon':
        // Names from http://another.example/
        case 'Uriah Heep':
        case 'David Copperfield':
        case 'Edward Murdstone':
            return 'Charles Dickens';

        // Names from http://further.example/
        case 'Nanki-Poo':
        case 'Pish-Tush':
        case 'Pooh-Bah':
            return 'W. S. Gilbert';
    }
}
```

Admittedly it is possible that a comment *might* be intended as a case body, and the lack of a break or fallthrough comment accidental. But in such cases, the comment would be expected to be indented beyond the `case` lines. I should say that at the very least if the comment is aligned with the `case` lines, it shouldn’t trigger `no-fallthrough`.

**Comment by not-an-aardvark** [comment]
Thanks for the report. This is actually working as intended (see https://github.com/eslint/eslint/issues/9095).

**Comment by jaydenseric** [comment]
I agree with OP, often when stacking cases you want to be able to comment what certain cases, or groups of cases represent. For grouping/legibility blank lines should also be allowed:

<img width="223" alt="screen shot 2018-06-15 at 10 45 08 pm" src="https://user-images.githubusercontent.com/1754873/41468595-e61df5cc-70ed-11e8-9d55-fc3711dfa97f.png">

Lots of us Prettier users only enable "recommended" rules that are meant to prevent bugs. As this rule is "recommended" it is surprising that it would be triggered by stylistic comment placement or whitespace. It's not intuitive for a comment or blank line to play the same role as case body code, as they are conceptually and practically totally different.

The original use case for the "works as intended" behavior, that people might comment out a case body and get surprised it changes the logic, is not very compelling IMHO. It should be assumed that people comment out what they intend to comment out. It's frustrating when a tool incorrectly assumes you are a fool and there is no elegant workaround.
